test/built-ins/TypedArrays/internals/DefineOwnProperty/key-is-not-canonical-index.js

```diff
@@ -0,0 +1,77 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-integer-indexed-exotic-objects-defineownproperty-p-desc
+description: |
+  Sets an ordinary property value if numeric key is not a CanonicalNumericIndex
+info: >
+  9.4.5.3 [[DefineOwnProperty]] ( P, Desc)
+  ...
+  3. If Type(P) is String, then
+    a. Let numericIndex be ! CanonicalNumericIndexString(P).
+    b. If numericIndex is not undefined, then
+    ...
+  4. Return OrdinaryDefineOwnProperty(O, P, Desc).
+  ...
+includes: [testTypedArray.js, propertyHelper.js]
+features: [Reflect]
+---*/
+
+var keys = [
+  "1.0",
+  "+1",
+  "1000000000000000000000",
+  "0.0000001"
+];
+
+var dataDesc = {
+  value: 42,
+  writable: true,
+  configurable: true
+};
+
+var fnset = function() {};
+var fnget = function() {};
+
+var acDesc = {
+  get: fnget,
+  set: fnset,
+  enumerable: true,
+  configurable: false
+};
+
+testWithTypedArrayConstructors(function(TA) {
+  keys.forEach(function(key) {
+    var sample = new TA();
+
+    assert.sameValue(
+      Reflect.defineProperty(sample, key, dataDesc),
+      true,
+      "return true after defining data property [" + key + "]"
+    );
+
+    assert.sameValue(sample[key], 42, "value is set to [" + key + "]");
+    verifyWritable(sample, key);
+    verifyConfigurable(sample, key);
+    verifyNotEnumerable(sample, key);
```

Comment:
`verifyConfigurable` is destructive, so this assertion will fail. Can you move it two lines up, just before `verifyWritable`?